### ISSUE TITLE
Title: Installation error with Python 3.12.4 on Ubuntu 24.04 (WSL on Windows 11): "urllib.error.HTTPError: HTTP Error 404: Not Found"

### ISSUE THREAD

**Issue by NyankoAndMamenoki** [python-poetry/install.python-poetry.org#142]
Description:
I encountered an error while trying to install Poetry using Python 3.12. The installation fails with the following error message:

```
urllib.error.HTTPError: HTTP Error 404: Not Found
```

However, I found that the installation succeeds when using Python 3.11.
Here are the steps to reproduce the issue:

Install Python 3.12
Run the Poetry installation command: python -m pip install poetry
The installation fails with the aforementioned error message

I have attached the full stacktrace below:

```
Installing Poetry (1.8.3): Creating environment
Traceback (most recent call last):
  File "<stdin>", line 314, in make
ModuleNotFoundError: No module named 'ensurepip'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 945, in <module>
  File "<stdin>", line 923, in main
  File "<stdin>", line 560, in run
  File "<stdin>", line 581, in install
  File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "<stdin>", line 653, in make_env
  File "<stdin>", line 639, in make_env
  File "<stdin>", line 340, in make
  File "/usr/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 559, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

Environment:

Operating System: Ubuntu 24.04 on Windows 11  (Windows Subsystem for Linux)
Python Version: 3.12.4
Poetry Version: 1.8.3

I suspect that there might be an issue with the Poetry installation script or its dependencies when running on Python 3.12. The installation works fine on Python 3.11, but fails on 3.12.
Please let me know if you need any additional information or if there are any workarounds or fixes available for this issue.
Thank you for your attention and assistance.

**Comment by edmorley** [comment]
Hi! 

This is because:
1. Ubuntu's system Python doesn't include the `ensurepip` module
2. When the Poetry installer cannot use the `ensurepip` module it falls back to using the `virtualenv` package by downloading it from the Python versioned URLs like `https://bootstrap.pypa.io/virtualenv/3.11/virtualenv.pyz` ([see here](https://github.com/python-poetry/install.python-poetry.org/blob/d62875fc05fb20062175cd14d19a96dbefa48640/install-poetry.py#L329-L333))
3. Ubuntu 24.04's system Python is 3.12, and the upstream `virtualenv` project hasn't published the versioned URL for Python 3.12 (ie `https://bootstrap.pypa.io/virtualenv/3.12/virtualenv.pyz` HTTP 404s)

I've filed:
https://github.com/pypa/virtualenv/issues/2744

And opened:
https://github.com/pypa/get-virtualenv/pull/46

**Comment by edmorley** [comment]
This issue can be closed, since it was fixed by:
https://github.com/pypa/get-virtualenv/pull/46
